core/sr-arithmetic/fuzzer/Cargo.toml

```diff
@@ -0,0 +1,20 @@
+[package]
+name = "sr-arithmetic-fuzzer"
+version = "2.0.0"
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
+
+[dependencies]
+honggfuzz = "0.5"
+sr-arithmetic = { path = ".." }
+primitive-types = "0.6"
+num-bigint = { git = "https://github.com/expenses/num-bigint", branch = "as-slice" }
```

Comment:
Since this is fuzz code, it is okay to depend on git dependencies, but is there any other way to not depend on this branch?

Reply:
Yeah, that's not quite ideal, but to compare the `BigUint` implementations I need access to the underlying slice, so I had to use my own branch. I'll try and make a PR for that.

Reply:
https://github.com/rust-num/num-bigint/pull/12 does what I want, but the repo seems to be moving pretty slowly.

Reply:
as fuzzer is part of workspace then the test-suite will fail if this branch is changed, I think this is not fine, not sure what the best way to solve this.

Reply:
This is used for the function `fn assert_biguints_eq(a: &BigUint, b: &num_bigint::BigUint)`
I think you can come up with a solution by converting `a` into num_bigint and use the partialeq implementation of num_bigint.

Or you can use the function https://docs.rs/num-bigint/0.2.3/num_bigint/struct.BigUint.html#method.to_radix_be or https://docs.rs/num-bigint/0.2.3/num_bigint/struct.BigUint.html#method.to_radix_le

Reply:
> > The fuzzing code found that this line:
> > https://github.com/paritytech/substrate/blob/0cd9bbfb721b5f0f31982f681bdf1e77612ecc79/core/sr-primitives/src/sr_arithmetic.rs#L1667
> > 
> > sometimes adds 1 to the value of the result when it shouldn't.
> 
> Actually, if you remove this line the fuzzer doesn't crash at all, so the issue here is that we want `multiply_by_rational` to round the value, but (probably?) not all the time.

Can you paste an example crash?

Reply:
Did you consider this? I suppose you can go to  `to_str_radix` and then easily convert every 4 chunks into a `Single`.

Reply:
Converting to `num_bigint::BigUint` seems like the easiest approach for now.